torch/distributions/constraints.py

```diff
@@ -297,6 +301,41 @@ def check(self, value):
         return (value == value).all()  # False for NANs.
 
 
+class _Cat(Constraint):
+    """
+    Constraint functor that applies a sequence of constraints
+    `cseq` at the submatrixes at dimension `dim`
+    in a way compatible with `torch.cat`.
+    """
+    def __init__(self, cseq, dim=0):
```

Comment:
nit: consider adding a line
```diff
  def __init__(self, cseq, dim=0):
+     cseq = list(cseq)
      ...
```
in case a user passes in a generator or mutable list for `cseq`